### PR TITLE
Fix TextField onChange bug

### LINKED ISSUE #10182: Controlled TextField component not accepting same (one character) input after clearing it
### Environment Information

- **Package version(s)**: 7.25.0
- **Browser and OS versions**: Chrome (latest)

### Please provide a reproduction of the bug in a codepen:

codesandbox reproduction:
[https://codesandbox.io/embed/nice-kirch-rldpx](https://codesandbox.io/embed/nice-kirch-rldpx)

#### Actual behavior:

When typing one character to the (controlled) input field and then hitting the button (recording the input value to an array and clearing it) the same character cannot be typed again into the field.

1. type the letter `a` to the input
2. hit record button
3. type the letter `a` to the input

#### Expected behavior:

Expected it to be the same behaviour as before clearing the input field.

### PR DESCRIPTION
**Pull request by ecraig12345**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #10182
- [x] Include a change request file using `$ yarn change`

#### Description of changes

TextField should reset the cached `_lastChangeValue` when the `value` prop changes. See #10182 for details on the scenario this fixes. 

Also added a lint rule banning `describe.only` and `it.only` from being checked in by accident.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/10194)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/TextField/TextField.test.tsx [hunk 1]
```diff
@@ -299,9 +307,23 @@ describe('TextField with error message', () => {
   it('should render error message when onGetErrorMessage returns a string', () => {
     const validator = jest.fn((value: string) => (value.length > 3 ? errorMessage : ''));
 
+    // function onNotifyResult(errorMessage: string | JSX.Element, value: string | undefined) {
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/TextField/TextField.test.tsx]: delete?

File: packages/office-ui-fabric-react/src/components/TextField/TextField.test.tsx [hunk 2]
```diff
@@ -299,9 +307,23 @@ describe('TextField with error message', () => {
...
+    //     done(ex);
+    //   }
+    // }
+
     wrapper = mount(<TextField onGetErrorMessage={validator} validateOnLoad={false} />);
 
     wrapper.find('input').simulate('input', mockEvent('also invalid'));
+    // // run the validation debounce timer
+    // jest.runOnlyPendingTimers();
+    // // // error message is inside a DelayedRender, so run that timer
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/TextField/TextField.test.tsx]: delete?

File: packages/office-ui-fabric-react/src/components/TextField/TextField.test.tsx [hunk 3]
```diff
@@ -314,6 +336,11 @@ describe('TextField with error message', () => {
     wrapper = mount(<TextField onGetErrorMessage={validator} validateOnLoad={false} />);
 
     wrapper.find('input').simulate('input', mockEvent('also invalid'));
+    // // run the validation debounce timer
+    // jest.runOnlyPendingTimers();
+    // // error message is inside a DelayedRender, so run that timer
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/TextField/TextField.test.tsx]: more

File: packages/tslint-rules/tslint.json [hunk 4]
```diff
@@ -180,6 +180,11 @@
       }
     ],
     "use-named-parameter": true,
+    "ban": [
+      true,
+      { "name": ["describe", "only"], "message": "describe.only should only be used during test development" },
```
Comment (dzearing) [inline packages/tslint-rules/tslint.json]: THANK YOU!

File: packages/office-ui-fabric-react/src/components/TextField/TextField.test.tsx (hunk 1 above)
Comment (ecraig12345) [reply packages/office-ui-fabric-react/src/components/TextField/TextField.test.tsx]: Oops, thought I pushed a change fixing this. Updated now.

**Pull request merged**